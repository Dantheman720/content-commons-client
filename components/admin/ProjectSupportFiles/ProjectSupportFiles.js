/**
 *
 * ProjectSupportFiles
 *
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';
import SupportFileTypeList from 'components/admin/projects/ProjectEdit/SupportFileTypeList/SupportFileTypeList';

import './ProjectSupportFiles.scss';

const ProjectSupportFiles = props => {
  const {
    heading,
    projectId,
    config
  } = props;


  const fileTypes = Object.keys( config );

  const renderFileTypes = type => (
    <Fragment key={ `file-type-${type}` }>
      <Grid.Column>
        <SupportFileTypeList
          projectId={ projectId }
          config={ config[type] }
        />

      </Grid.Column>
    </Fragment>
  );

  return (
    <div className="support-files">
      <Grid stackable>
        <Grid.Row>
          <Grid.Column width={ 16 }>
            <h2 className="heading uppercase">{ heading }</h2>
          </Grid.Column>
        </Grid.Row>

        <Grid.Row columns={ fileTypes.length } divided>
          { fileTypes.map( renderFileTypes ) }
        </Grid.Row>

      </Grid>
    </div>
  );
};

ProjectSupportFiles.propTypes = {
  heading: PropTypes.string,
  projectId: PropTypes.string,
  config: PropTypes.object.isRequired,
};

export default ProjectSupportFiles;
