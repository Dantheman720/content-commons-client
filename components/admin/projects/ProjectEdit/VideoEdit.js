import React from 'react';

const VideoEdit = () => (
  <div>Video Edit</div>
);

export default VideoEdit;
