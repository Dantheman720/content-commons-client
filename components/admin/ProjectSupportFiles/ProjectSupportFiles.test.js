import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { projects, supportFilesConfig } from 'components/admin/ProjectEdit/mockData';
import ProjectSupportFiles from './ProjectSupportFiles';

const props = {
  heading: 'Support Files',
  projectId: { videoID: 'made-in-america' },
  supportFiles: projects[0].supportFiles,
  hasSubmittedData: false,
  protectImages: true,
  handleChange: jest.fn(),
  config: supportFilesConfig,
  hasUploaded: true
};

const Component = <ProjectSupportFiles { ...props } />;

describe( '<ProjectSupportFiles />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );
    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'renders the heading', () => {
    const wrapper = shallow( Component );
    const heading = wrapper.find( '.heading' );
    expect( heading.name() ).toEqual( 'h2' );
    expect( heading.text() ).toEqual( props.heading );
  } );

  it( 'renders the correct number of `SupportFileTypeList`s', () => {
    const wrapper = shallow( Component );
    const lists = wrapper.find( 'SupportFileTypeList' );
    const typesCount = Object.keys( props.config ).length;
    expect( lists ).toHaveLength( typesCount );
  } );

  it( 'renders a `Checkbox` for protect images if the type is `other`', () => {
    const wrapper = shallow( Component );

    expect( wrapper.find( 'Checkbox' ).exists() )
      .toEqual( true );
    wrapper.setProps( {
      config: {
        srt: {
          headline: 'Test Files',
          fileType: 'abc',
          popupMsg: 'A test message'
        }
      }
    } );
    expect( wrapper.find( 'Checkbox' ).exists() )
      .toEqual( false );
  } );

  it( 'checking/unchecking `Checkbox` calls `handleChange`', () => {
    const wrapper = shallow( Component );
    const checkbox = wrapper.find( 'Checkbox' );
    if ( checkbox.exists() ) {
      checkbox.simulate( 'change' );
      expect( props.handleChange ).toHaveBeenCalled();
    }
  } );

  it( 'does not render if no support files', () => {
    const wrapper = shallow( Component );
    wrapper.setProps( { supportFiles: {} } );
    expect( wrapper.find( '.support-files' ) )
      .toHaveLength( 0 );
  } );
} );
