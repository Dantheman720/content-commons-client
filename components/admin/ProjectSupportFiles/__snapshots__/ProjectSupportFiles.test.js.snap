// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectSupportFiles /> renders without crashing 1`] = `
<div
  className="support-files"
>
  <Grid
    stackable={true}
  >
    <GridRow>
      <GridColumn
        width={16}
      >
        <h2
          className="heading uppercase"
        >
          Support Files
        </h2>
      </GridColumn>
    </GridRow>
    <GridRow
      columns={2}
      divided={true}
    >
      <GridColumn>
        <SupportFileTypeList
          config={
            Object {
              "types": Object {
                "other": Object {
                  "editTitle": "Edit thumbnail files in this project",
                  "extensions": Array [
                    ".png",
                    ".jpeg",
                    ".jpg",
                  ],
                  "headline": "Additional Files",
                  "iconMsg": "Checking this prevents people from downloading and using your images. Useful if your images are licensed.",
                  "iconSize": "small",
                  "iconType": "info circle",
                  "popupMsg": "Additional files that can be used with this video, e.g., audio file, pdf.",
                },
                "srt": Object {
                  "editTitle": "Edit srt files in this project",
                  "extensions": Array [
                    ".srt",
                  ],
                  "headline": "SRT Files",
                  "popupMsg": "Some info about what SRT files are.",
                },
              },
            }
          }
          projectId="123"
          save={[MockFunction]}
          type="srt"
        />
      </GridColumn>
      <GridColumn>
        <SupportFileTypeList
          config={
            Object {
              "types": Object {
                "other": Object {
                  "editTitle": "Edit thumbnail files in this project",
                  "extensions": Array [
                    ".png",
                    ".jpeg",
                    ".jpg",
                  ],
                  "headline": "Additional Files",
                  "iconMsg": "Checking this prevents people from downloading and using your images. Useful if your images are licensed.",
                  "iconSize": "small",
                  "iconType": "info circle",
                  "popupMsg": "Additional files that can be used with this video, e.g., audio file, pdf.",
                },
                "srt": Object {
                  "editTitle": "Edit srt files in this project",
                  "extensions": Array [
                    ".srt",
                  ],
                  "headline": "SRT Files",
                  "popupMsg": "Some info about what SRT files are.",
                },
              },
            }
          }
          projectId="123"
          save={[MockFunction]}
          type="other"
        />
      </GridColumn>
    </GridRow>
  </Grid>
</div>
`;
